### PR TITLE
[2840] Added redirect for the wizard subject page

### PR DESCRIPTION
**Pull request by defong**
### Context
Redirects to rails

### Changes proposed in this pull request
Added redirect for the wizard subject page

### Guidance to review

Set these

```
FEATURE_REDIRECT_TO_RAILS
NEW_APP_URL
```

This endpoint 
`/start/subject`
will be redirected along with query strings where appropriate

### CODE REVIEW

**Comment by defong** [comment]
[subjects start: Redirect from C# to Rails](https://trello.com/c/APzEzitF/2785-subjects-start-redirect-from-c-to-rails)

File: tests/Unit.Tests/Controllers/FilterControllerTests.cs [hunk 1]
```diff
@@ -146,5 +148,32 @@ public void QualificationGetRedirectsToNewApp()
...
         }
+
+        [Test]
+        public void SubjectWizardGetHttpGetTest()
+        {
+            var result = _filterController.SubjectWizardGet(_resultsFilter);
+            result.Should().BeOfType<ViewResult>();
+            var viewResult = result as ViewResult;
+            viewResult?.Model.Should().BeOfType(typeof(SubjectFilterViewModel));
+            _mockApi.Verify(x => x.GetSubjectAreas(), Times.AtMostOnce);
```
Comment (timabell) [inline tests/Unit.Tests/Controllers/FilterControllerTests.cs]: why `AtMostOnce`? Surely you know how many times it would be called?

Comment (defong) [reply tests/Unit.Tests/Controllers/FilterControllerTests.cs]: fixed

File: tests/Unit.Tests/Controllers/FilterControllerTests.cs [hunk 2]
```diff
@@ -146,5 +148,32 @@ public void QualificationGetRedirectsToNewApp()
...
+            _mockFlag.Setup(x => x.RedirectToRails).Returns(true);
+            string actualRedirectPath = "start/subject?query";
+
+            var redirectObject = new RedirectResult(actualRedirectPath);
+            _redirectUrlMock.Setup(x => x.RedirectToNewApp())
+                .Returns(redirectObject);
+
+            var result = _filterController.SubjectWizardGet(_resultsFilter) as RedirectResult;
+            result.Should().Be(redirectObject);
+            _redirectUrlMock.Verify(x => x.RedirectToNewApp(), Times.AtLeastOnce);
```
Comment (timabell) [inline tests/Unit.Tests/Controllers/FilterControllerTests.cs]: can you be more precise than `AtLeastOnce`

Comment (defong) [reply tests/Unit.Tests/Controllers/FilterControllerTests.cs]: fixed

**Comment by defong** [comment]
superseded by #509

**Pull request closed without merging**